### PR TITLE
make more specific signatures around teams.Get()

### PR DESCRIPTION
**Pull request by maxtaco**
- also, fix `sigs list` which was broken for `eldest` and `sibkey` types

### CODE REVIEW

File: go/libkb/id_table.go [hunk 1]
```diff
@@ -618,9 +618,13 @@ func (s *EldestChainLink) Type() string                  { return string(Delegat
 func (s *EldestChainLink) ToDisplayString() string       { return s.kid.String() }
 func (s *EldestChainLink) GetDevice() *Device            { return s.device }
 func (s *EldestChainLink) GetPGPFullHash() string        { return s.extractPGPFullHash("key") }
+func (s *EldestChainLink) insertIntoTable(tab *IdentityTable) {
```
Comment (maxtaco) [inline go/libkb/id_table.go]: @oconnor663 this broke `keybase sigs list`, fixed with a specific implementation that wasn't a generic.

File: go/libkb/id_table.go [hunk 2]
```diff
@@ -664,6 +668,9 @@ func (s *SibkeyChainLink) Type() string                  { return string(Delegat
 func (s *SibkeyChainLink) ToDisplayString() string       { return s.kid.String() }
 func (s *SibkeyChainLink) GetDevice() *Device            { return s.device }
 func (s *SibkeyChainLink) GetPGPFullHash() string        { return s.extractPGPFullHash("sibkey") }
+func (s *SibkeyChainLink) insertIntoTable(tab *IdentityTable) {
+	tab.insertLink(s)
+}
```
Comment (maxtaco) [inline go/libkb/id_table.go]: @oconnor663  ditto

**Comment by mlsteele** [comment]
KBFS and Chat both will want different properties in their calls for encrypting and decrypting. For encrypting they want to hit the cache unless it's too stale. But for decrypting they want to send NeedKeyGeneration and WantMembers.

File: go/protocol/keybase1/extras.go [hunk 3]
```diff
@@ -1317,7 +1317,7 @@ func (k TeamApplicationKey) Material() Bytes32 {
 }
 
 func (k TeamApplicationKey) Generation() int {
-	return k.KeyGeneration
+	return int(k.KeyGeneration)
```
Comment (mlsteele) [inline go/protocol/keybase1/extras.go]: return new type?

Comment (maxtaco) [inline go/protocol/keybase1/extras.go]: can't, it breaks an interface.

**Comment by keybase-ci-visdiff** [comment]
The commits d0a3a0e447a7...f6408adb6199 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Tracker-New_user_Scroll_1** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1497986254074-d0a3a0e447a7-f6408adb6199-linux/Tracker-New_user_Scroll_1-d0a3a0e447a7.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1497986254074-d0a3a0e447a7-f6408adb6199-linux/Tracker-New_user_Scroll_1-f6408adb6199.png)  
   Tracker-New_user_Scroll_1 comparison
</summary>

**Comment by maxtaco** [comment]
I have a feeling we'll be changing this constantly 

;)

**Comment by keybase-ci-visdiff** [comment]
The commits d0a3a0e447a7...8460c6f4c1af introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Tracker-New_user_Scroll_1** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1497987052811-d0a3a0e447a7-8460c6f4c1af-linux/Tracker-New_user_Scroll_1-d0a3a0e447a7.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1497987052811-d0a3a0e447a7-8460c6f4c1af-linux/Tracker-New_user_Scroll_1-8460c6f4c1af.png)  
   Tracker-New_user_Scroll_1 comparison
</summary>

**Pull request merged**